assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -393,61 +273,48 @@ pub fn u32rotr(
+            // Assert the top of the stack is a u32 value.
+            // NOTE: We cannot use U32Assert2 since we are potentially pushing a number larger
+            // than u32 for b.
+            // TODO: double check this logic: we shouldn't use U32MUL if both inputs are not
+            // guaranteed to be u32 values
+            validate_param(imm, 0, 31)?;
+            span.push_ops([Pad, U32assert2, Drop, Push(Felt::new(1 << (32 - imm)))]);
+        }
+        (Some(imm), U32OpMode::Unchecked) => {
+            validate_param(imm, 0, 31)?;
+            span.push_op(Push(Felt::new(1 << (32 - imm))));
+        }
+        (None, U32OpMode::Checked) => {
+            span.push_ops([
                 // Verify both b and a are u32.
-                span.push_op(U32assert2);
+                U32assert2,
                 // Calculate 32 - b and assert that the shift value b <= 31.
-                span.push_op(Push(Felt::new(31)));
-                span.push_op(Dup1);
-                span.push_op(U32sub);
-                span.push_op(Not);
-                span.push_op(Assert);
-                span.push_op(Incr);
-                span.push_op(Dup1);
+                Push(Felt::new(31)),
+                Dup1,
+                U32sub,
+                Not,
+                Assert,
+                Incr,
+                Dup1,
                 // If 32-b = 32, replace it with 0.
-                span.push_op(Eqz);
-                span.push_op(Not);
-                span.push_op(CSwap);
-                span.push_op(Drop);
-                append_pow2_op(span);
-                span.push_op(Swap);
-            }
-            U32OpMode::Unchecked => {
-                span.push_op(Push(Felt::new(32)));
-                span.push_op(Swap);
-                span.push_op(U32sub);
-                span.push_op(Drop);
-                append_pow2_op(span);
-            }
-            _ => unreachable!("unsupported operation mode"),
-        },
-        Some(imm) => match op_mode {
-            U32OpMode::Checked => {
-                // Assert the top of the stack is a u32 value.
-                // NOTE: We cannot use U32Assert2 since we are potentially pushing a number larger
-                // than u32 for b.
-                // TODO: double check this logic: we shouldn't use U32MUL if both inputs are not
-                // guaranteed to be u32 values
-                span.push_op(Pad);
-                span.push_op(U32assert2);
-                span.push_op(Drop);
-
-                validate_param(imm, 0, 31)?;
-                span.push_op(Push(Felt::new(2u64.pow(32 - imm as u32))));
-            }
-            U32OpMode::Unchecked => {
-                span.push_op(Push(Felt::new(2u64.pow(32 - imm as u32))));
-            }
-            _ => unreachable!("unsupported operation mode"),
-        },
+                Eqz,
+                Not,
+                CSwap,
+                Drop,
+            ]);
```

Comment:
Similar to my other comment: I would probably split this into two separate `push_ops()` statements so that rustfmt doesn't put each operation onto a separate line.

Reply:
Fixed